_posts/2018-08-17-snippets-brotli.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: post
+title: "Using Brotli compression to reduce CDN costs"
+author: Giorgos Logiotatidis
+author_url: https://giorgos.sealabs.net
+excerpt_separator: <!--more-->
+---
+
+The Snippets Service allows Mozilla to communicate with Firefox users directly
+by placing a snippet of text and an image on their homepage. Snippets share
+exciting news from the Mozilla World, useful tips and tricks based on user
+activity and sometimes jokes.
+
+To achieve personalized, activity based messaging in a privacy respecting and
+efficient manner, the service creates a Bundle of Snippets per locale. The
+complete Bundle is transferred to the client where a locally executed algorithm
+decides which snippet to display. A carefully designed system with multiple
+levels of caching takes care of the delivery. One layer of caching is CloudFront
+CDN.
+
+
+## The problem
+
+During the last months we observed a significant uptake of our CDN costs as the
+Mozilla's Lifecycle Marketing Team was increasing the number of Snippets for the
+English language from about 10 to 150.
+
+The Bundle file-size increased from about 200 KiB to more than 4MiB, which
+translated to more bytes transferred using the CDN and thus higher bills. Given
+that Firefox requests new bundles every 4 hours, this quickly became a issue
```

Comment:
I would love to see some mention of the scale of snippets. So the interesting thing here is that snippets is grabbed every 4 hours by every client. I think it would be cool to mention some numbers.. big numbers!